zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/distribution/CommandDistributionBehavior.java

```diff
@@ -187,6 +191,23 @@ void distributeNextInQueue(final long finishedDistributionKey, final int partiti
                     nextDistributionKey));
   }
 
+  private void continueAfterQueue(final String queue) {
+    if (distributionState.hasQueuedDistributions(queue)) {
+      return;
+    }
+    distributionState.forEachContinuationCommand(queue, this::handleContinuationCommand);
+  }
+
+  private void handleContinuationCommand(final long key, final CommandDistributionRecord record) {
+    final var intent = record.getIntent();
+    final var value = record.getCommandValue();
+
+    commandWriter.appendFollowUpCommand(key, intent, value);
```

Comment:
❌ We don't know how much continuation commands are in our state and how large the record values are. This seems like a big risk to exceed the maximum batch size. I think we'd be better off writing a `CONTINUE` command with just the queueId and the key. That way we can write all the actual commands in separate batches.

Reply:
Writing this comment makes me realize that the continuation naming still confuses me a bit. If we add a  `CommandDistributionIntent.CONTINUE` command, what will it continue. Will it write the next command in the queue, or will it take care of the callback. Hence why I think callback is a better name.

Reply:
True, I haven't thought about that :+1: 
To avoid confusing naming we could do this:
1. On ACK, when the queue is found to be empty, write `CommandDistributionIntent.FINISH_QUEUE` command.
2. When processing `CommandDistributionIntent.FINISH_QUEUE` write x continuation commands and matching `CommandDistributionIntent.CONTINUATION_COMPLETED` events.
3. If we still have remaining continuation commands at the end of handling `CommandDistributionIntent.FINISH_QUEUE`, write another `CommandDistributionIntent.FINISH_QUEUE` command.
4. If we don't have continuation commands at the end of handling `CommandDistributionIntent.FINISH_QUEUE`, write a no-op `CommandDistributionIntent.FINISHED_QUEUE` event.

Something like that?

Reply:
On the other hand this complicates the implementation a good bit. I'm almost inclined to just remove the feature to register multiple continuation commands for that reason. If we only have one continuation command, we don't need to care about batch size at all.

Reply:
> On the other hand this complicates the implementation a good bit. I'm almost inclined to just remove the feature to register multiple continuation commands for that reason. If we only have one continuation command, we don't need to care about batch size at all.

I think that's fair, let's not over-engineer this until we actually need it. 👍

Reply:
Because removing this functionality made implementation slightly more difficult, I've opted for you suggestion to first write small `CONTINUE` commands which in turn write the continuation command and the `CONTINUATION_COMPLETED` event. This should avoid the batch size issue except in pathological cases where there's thousands of continuation commands and we can't even write all of the `CONTINUE` commands.